aten/src/ATen/native/cuda/ForeachFunctors.cuh

```diff
@@ -18,13 +19,13 @@ inline void increment_version(TensorList tensors) {
 }
 
 // Initializes args and checks if all args are aligned
-template <int depth, typename T>
+template <int depth, typename T, typename index_t>
 __device__ bool init_args(
     T** args,
     TensorListMetadata<depth>& tl,
-    const int64_t chunk_idx,
-    const int64_t chunk_size,
-    const int64_t tensor_loc) {
+    const index_t chunk_idx,
+    const int chunk_size,
+    const int tensor_loc) {
```

Comment:
`chunk_size` is always the costexpr var of `kChunkSize` defined in `MultiTensorApply.cuh`. I find it worth thinking of removing this argument but it'd break any cuda extensions of multi_tensor_apply`.

`tensor_loc` IIUC is basically `unsigned char`.